### PR TITLE
React Context API blog post

### PR DESCRIPTION
**Pull request by eylonronen**
#### Please note that in case of below checklist not updated accordingly, the maintainers have the right to immediately close your pull request.

- [x] You created a branch from `dev` branch.
- [x] You PR is raised on `dev` branch and not on`master`.
- [x] Blog follows [Blog Guidelines](https://github.com/LoginRadius/engineering-portal/blob/master/GUIDELINES.md).
- [x] You have read the [Contribution Guidlines](https://github.com/LoginRadius/engineering-portal/blob/master/CONTRIBUTING.md) before creating this PR.
- [x] Checked for any duplicate blog entry.
- [x] Checked for all grammatical errors in the blog.
- [x] You have not published the blog content anywhere else on the internet on platforms like medium, dev.to etc.
- [x] The content is not copy-pasted from anywhere else on the Internet.

### CODE REVIEW

File: content/blog/react-contedxt-api/index.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
+---
+title: "What Is React Context API and How To Use It Best"
+date: "2020-10-05"
+coverImage: "react.png"
+author: "Eylon Ronen"
+tags: ["React", "Redux", "State Management", "React Hooks", "React Higher Order Component"]
+---
+
+## What is Context API?
```
Comment (vyasriday) [inline content/blog/react-contedxt-api/index.md:9]: The sentence should be ended with a full stop, right?

File: content/blog/react-contedxt-api/index.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+Context API is a (kinda) new feature added in version 16.3 of React that allows one to manage state lightly and with ease
+
+### How it works
+React.createContext() is all you need. It returns a consumer and a provider.
+**Provider** is a component that as it names suggests provides the state to its children.
```
Comment (vyasriday) [inline content/blog/react-contedxt-api/index.md:14]: should be `as it's name suggests`

File: content/blog/react-contedxt-api/index.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+Context API is a (kinda) new feature added in version 16.3 of React that allows one to manage state lightly and with ease
+
+### How it works
+React.createContext() is all you need. It returns a consumer and a provider.
+**Provider** is a component that as it names suggests provides the state to its children.  
+**Consumer** as it so happens is a component that consumes and uses the state.
+
+### **Does it replace Redux?**
+No... Well not entirely.  
+Redux is great and came perfectly to answer the need for state management. actually it answered this need so well that it came to be known that you can't be a "true" React developer if you don't know your way around Redux.
```
Comment (vyasriday) [inline content/blog/react-contedxt-api/index.md:19]: start sentence with a capital.  `Actually it answered this need . . . `

File: content/blog/react-contedxt-api/index.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+However Redux has its disadvantages and that's why it's important to know what Context API gives us that Redux doesn't:
+* Simplicity - When using redux people tend to manage almost all of their state in redux and it causes 2 problems:  
+  1. Overhead - Why should I create/update 3 files just to add one tiny feature? 
+  2. One of the great advantages of React's one-way data binding is that it's easy to understand - A component passes state to its children. Using Redux takes it away from us
```
Comment (vyasriday) [inline content/blog/react-contedxt-api/index.md:23]: End sentence with a full stop

File: content/blog/react-contedxt-api/index.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+* Using Context API we can define several unrelated contexts (stores) and use each in it's proper place in the app.
+
+## How to use the best
+Well I'm convinced. How am I implementing Context API in my app?
+First, make sure you need it. sometimes people use shared state across nested components instead of just passing it as props.
```
Comment (vyasriday) [inline content/blog/react-contedxt-api/index.md:28]: A sentence must start with a capital : `Sometimes people use . . . `

**Review by vyasriday — CHANGES_REQUESTED** [review]

**Review by puneetsingh24 — CHANGES_REQUESTED** [review]
Hello @eylonronen, Thanks for writing this blog and we appreciate your effort while writing this. However, as per our project contribution guidelines, a blog should contain minimum of 600 words excluding meta description on top.

Please add more information about react context API

**Review by vyasriday — APPROVED** [review]

**Review by puneetsingh24 — APPROVED** [review]

**Comment by vyasriday** [comment]
@eylonronen Hey can you provide us with some details like your name, Linkedin and all to show it on your profile for our engineering blog?

**Comment by eylonronen** [comment]
name: Eylon Ronen
linkedin: https://www.linkedin.com/in/eylon-ronen-366916157

**Pull request merged**